src/model/geom/track.py

```diff
@@ -41,12 +42,50 @@ def segment_walls(self, segment_id: SegmentId) -> List[Wall]:
 
         return walls
 
-    def sense_closest(self, sensor: Sensor, active_segment: SegmentId) -> float:
-        """Return the distance to the closest wall of the track sensed with the given sensor."""
-        # TODO: edge cases when a later segment has the closest wall
-        # TODO: optimize for multiple sensors
-        for segment_id in utils.spread_int(active_segment, 0, len(self.segments)):
+    def _spread_segment_ids(self, segment_id: SegmentId) -> Iterable[SegmentId]:
+        return utils.spread_int(segment_id, 0, len(self.segments))
+
+    def sense_closest(
+        self, sensors: List[Sensor], active_segment: SegmentId
+    ) -> List[float]:
+        """Returns the distance to the closest wall of the track sensed with the given sensor."""
+        # TODO?: edge cases when a later segment has
+        #  the closest wall (not sure if that ever happens)
```

Comment:
I was thinking about it before and I can't imagine this situation, but also I'm not able to present mathematical proof.